tests/test_functional.py

```diff
@@ -1,49 +1,18 @@
-from unittest.mock import Mock
-from collections import namedtuple
 import pytest
 
 
 from datenguide_python.query_execution import QueryExecutioner
+from datenguide_python.query_builder import Query, Field
 
 
 @pytest.fixture
-def sample_queries():
-    q1 = Mock()
-    q1.get_graphql_query.return_value = """
-    {
-        region(id:"05911") {
-            id
-            name
-            BEVMK3 {
-                value
-                year
-            }
-        }
-    }
-    """
-    q1.get_fields.return_value = ["region", "id", "name", "BEVMK3", "value", ".year"]
-
-    mq1 = Mock()
-    mq1.get_graphql_query.return_value = """
-    {
-      __type(name: "Region") {
-        name
-        fields {
-          name
-          description
-          type {
-            name
-            kind
-          }
-        }
-      }
-    }
-    """
-    SampleQueries = namedtuple("SampleQueries", "data_query1 meta_query1")
-    return SampleQueries(q1, mq1)
-
-
-def test_QueryExecutionerWorkflow(sample_queries):
+def query():
+    field = Field(name="BEVMK3", fields=["value", "year"])
+    query = Query(region="05911", fields=["id", "name", field])
```

Comment:
Ideally define this the other way around, as the functional test case should mimic the intended use by the end user. I imagine this to be
```python
query = Query.regionQuery('05911')
query.add_field('id') # these could probably be added as defaults in the factory method
query.add_field('name')
statistic = query.add_field('BEVMK3')
statistic.add_field('value') # These should probably also be added as defaults together with source
statistic.add_field('year')
```

As mentioned in the snipped some defaults subfields might make sense. This is easy to do for the factory methods and a little more tricky for the statistics, but should also be possible using the meta data, and can easily be done in a future step.

Reply:
i would make both ways possible. Add fields with creation of query or add afterwards

Reply:
This is a subtle point anyway, as we are only talking about the structure of the functional test. But I would agree that the user should have the ability to use both ways. In turn this means that there should also be tests for both cases but I think that is covered with the unit tests. So I think it's fine.